utils/markdown.go

```diff
@@ -0,0 +1,179 @@
+func (r *notificationRenderer) renderDefault(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderHeading(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	if !entering {
+		if node.NextSibling() != nil {
+			_, _ = w.WriteString(" ")
+		}
+	}
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderListItem(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	if entering {
+		fc := node.FirstChild()
+		if fc != nil {
+			if _, ok := fc.(*ast.TextBlock); !ok {
+				_ = w.WriteByte(' ')
+			}
+		}
+	} else {
+		if node.NextSibling() != nil {
+			_, _ = w.WriteString(" ")
+		}
+	}
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderParagraph(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	if !entering {
+		if node.NextSibling() != nil {
+			_, _ = w.WriteString(" ")
+		}
+	}
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderCodeBlock(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	n := node.(*ast.CodeBlock)
+	source = bytes.ReplaceAll(source, []byte("\n"), []byte(" "))
+	if entering {
+		r.writeLines(w, source, n)
+	}
+
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderFencedCodeBlock(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	n := node.(*ast.FencedCodeBlock)
+	source = bytes.ReplaceAll(source, []byte("\n"), []byte(" "))
+	if entering {
+		r.writeLines(w, source, n)
+	}
+
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderText(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	if !entering {
+		return ast.WalkContinue, nil
+	}
+	n := node.(*ast.Text)
+	segment := n.Segment
+	if n.IsRaw() {
+		_, _ = w.Write(segment.Value(source))
+	} else {
+		_, _ = w.Write(segment.Value(source))
+		if n.HardLineBreak() || n.SoftLineBreak() {
+			_ = w.WriteByte(' ')
+		}
+	}
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderTextBlock(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	if !entering {
+		if _, ok := node.NextSibling().(ast.Node); ok && node.FirstChild() != nil {
+			_ = w.WriteByte(' ')
+		}
+	}
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderString(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	if !entering {
+		return ast.WalkContinue, nil
+	}
+	n := node.(*ast.String)
+	_, _ = w.Write(n.Value)
+	return ast.WalkContinue, nil
+}
+
+func (r *notificationRenderer) renderTable(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	return ast.WalkSkipChildren, nil
+}
```

Comment:
Any reason why tables are not rendered properly? It is okay to just let them be markdown and revisit it in a next PR. But this just skips the rendering entirely.

Reply:
![Screenshot_1624691285](https://user-images.githubusercontent.com/4019710/123505201-f5574b80-d687-11eb-806f-16ce828bfbb6.png)

Maybe this still acceptable for user read it.

Reply:
I think this is a good start. Tables are not that common. We can come back to it later.

Reply:
okay, I will return original string for tables.